hack/update-vendor.sh

```diff
@@ -187,134 +234,179 @@ group_replace_directives
+#    echo "=== tidying ${repo}" >> "${LOG_FILE}"
+#
+#    # prune replace directives that pin to the naturally selected version.
+#    # do this before tidying, since tidy removes unused modules that
+#    # don't provide any relevant packages, which forgets which version of the
+#    # unused transitive dependency we had a require directive for,
+#    # and prevents pruning the matching replace directive after tidying.
+#    go list -m -json all |
+#      jq -r 'select(.Replace != null) |
+#             select(.Path == .Replace.Path) |
+#             select(.Version == .Replace.Version) |
+#             "-dropreplace \(.Replace.Path)"' |
+#    xargs -L 100 go mod edit -fmt
+#
+#    go mod tidy -v >>"${LOG_FILE}" 2>&1
+#
+#    # disallow transitive dependencies on k8s.io/kubernetes
+#    loopback_deps=()
+#    kube::util::read-array loopback_deps < <(go list all 2>/dev/null | grep k8s.io/kubernetes/ || true)
+#    if [[ -n ${loopback_deps[*]:+"${loopback_deps[*]}"} ]]; then
+#      kube::log::error "Disallowed ${repo} -> k8s.io/kubernetes dependencies exist via the following imports:
+#$(go mod why "${loopback_deps[@]}")"
+#      exit 1
+#    fi
+#
+#    # prune unused pinned replace directives
+#    comm -23 \
+#      <(go mod edit -json | jq -r '.Replace[] | .Old.Path' | sort) \
+#      <(go list -m -json all | jq -r .Path | sort) |
+#    xargs -L 1 -I {} echo "-dropreplace={}" |
+#    xargs -L 100 go mod edit -fmt
+#
+#    # prune replace directives that pin to the naturally selected version
+#    go list -m -json all |
+#      jq -r 'select(.Replace != null) |
+#             select(.Path == .Replace.Path) |
+#             select(.Version == .Replace.Version) |
+#             "-dropreplace \(.Replace.Path)"' |
+#    xargs -L 100 go mod edit -fmt
+#
+#  popd >/dev/null 2>&1
+#done
+#echo "=== tidying root" >> "${LOG_FILE}"
+#go mod tidy >>"${LOG_FILE}" 2>&1
+#
+#
+## Phase 6: add generated comments to go.mod files
+#kube::log::status "go.mod: adding generated comments"
+#add_generated_comments "
+#// This is a generated file. Do not edit directly.
+#// Run hack/pin-dependency.sh to change pinned dependency versions.
+#// Run hack/update-vendor.sh to update go.mod files and the vendor directory.
+#"
+#for repo in $(kube::util::list_staging_repos); do
+#  pushd "staging/src/k8s.io/${repo}" >/dev/null 2>&1
+#    add_generated_comments "// This is a generated file. Do not edit directly."
+#  popd >/dev/null 2>&1
+#done
+#
 
 # Phase 6: rebuild vendor directory
 
 kube::log::status "vendor: running 'go mod vendor'"
+rm -rf staging-kube
+mv vendor/k8s.io/kubernetes/staging staging-kube
 go mod vendor >>"${LOG_FILE}" 2>&1
+for repo in $(list_staging_repos staging-kube/src/k8s.io); do
+  cp "staging-kube/src/k8s.io/${repo}/go.mod" "vendor/k8s.io/kubernetes/staging/src/k8s.io/${repo}"
+done
+rm -rf staging-kube
+ln -fs vendor/k8s.io/kubernetes/staging staging-kube
+
+# Phase 7: unprune kubernetes by checking it out completely
```

Comment:
I wasn't looking really closely, but can we just unprune everything?

Reply:
if you add **all** packages to the kube.go, sure ;-)